src/client/storage/uploader.ts

```diff
@@ -6,6 +6,11 @@ import {promises as fs} from 'fs';
 import path from 'path';
 import config from 'config';
 import logger from '../../core/log';
+import {storage} from './storage';
+import {eachLimit} from 'async';
+
+const MAX_UPLOAD_CONCURRENCY = 100;
+const MAX_VALIDATION_CONCURRENCY = 100;
```

Comment:
This should not be hard-coded here. Please add to `config/default.yaml` and `config/custom_env_variables.yaml` instead. There is a property `storage.concurrent_upload_limit` already. I think the `CONCURRENT_UPLOAD_LIMIT` variable is also used for that.